br/pkg/lightning/restore/restore.go

```diff
@@ -1442,12 +1425,26 @@ func (rc *Controller) restoreTables(ctx context.Context) error {
 
 	periodicActions, cancelFunc := rc.buildRunPeriodicActionAndCancelFunc(ctx, stopPeriodicActions)
 	go periodicActions()
-	finishFuncCalled := false
+
 	defer func() {
-		if !finishFuncCalled {
-			finishSchedulers()
-			cancelFunc(switchBack)
-			finishFuncCalled = true
+		finishSchedulers()
```

Comment:
Please move `defer close(stopPeriodicActions)` before this defer to ensure output progress log in post-process phase. BTW, I'm not sure if do normal logic like postProcess in a defer function is a good idea. @kennytm what do you think

Reply:
>  I'm not sure if do normal logic like postProcess in a defer function is a good idea.

Maybe it is not a good idea. 🤣 `restoreTables`  method need to do some refactor.

Reply:
> move defer close(stopPeriodicActions) before this defer to ensure output progress log in post-process phase

If so,  we can't switch TiKV to normal mode during duplicate detection.

Reply:
So, maybe we need to split stopPeriodicActions into two function, one is called at this point, and one is called after restoreTables finished. 😂

Reply:
It seems cancelFuncs did such thing.